### ISSUE TITLE
Regional Page Additional Tabs 

### ISSUE THREAD

**Issue by mmusori** [IFRCGo/go-frontend#1621]
**Is your feature request related to a problem? Please describe.**
The Regional pages only have three predefined  in tabs for addition information. The additional information tab is the only place that Regions can input all the different information that they have. for exmaple the Americas pagehttps://go.ifrc.org/regions/1#additional-info

![image](https://user-images.githubusercontent.com/7780809/94024899-3a2c1a00-fdb8-11ea-9a6c-7a36262acf7e.png)

**Describe the solution you'd like**
 We would like to have 3 additional tabs for the Regions to define and add in varied  information they have. The feature is already built into the Emergency pages.
![image](https://user-images.githubusercontent.com/7780809/94025556-ea9a1e00-fdb8-11ea-95f1-834a629033a6.png)

 And a good example of the use of the additional tabs on the emergency pages is:
https://go.ifrc.org/emergencies/3972#details

**Comment by mariam-yu** [comment]
@mmusori hope it's ok, I moved this task to the design stage. Please see below the links to the content mapping for the regional pages (based on currently information in Additional Data section and Africa Ops Room) and the wireframes.

Content mapping: https://docs.google.com/spreadsheets/d/1anuNStxy8vMxHl6voJXguSseXYWe2cZYuoNPSopaW5Q/edit#gid=0

Wireframes: https://invis.io/ZBZB8SF9UCP#/436634656_2-1_GO_Regions_AF (same password)

You can toggle between all the regions from the top nav. We created pages for all the regions to highlight where the iframes may come in, splitting the current Additional data pages. Check especially Africa and Americas regions as they had more content. @batpad @mmusori please see the suggestion for the Ops Room split, which is mainly in the Regional Profile (additional links) and preparedness. 

Let me know if you have any questions.

**Comment by batpad** [comment]
Tagging @kamicut and @imohkay here - these are in the process of being finalized - will be good to take a look and then discuss -- the broad idea here is to allow users in the Admin to add custom Iframed data to the Region and Country pages.

To avoid things being too arbitrarily fragmented, @mariam-yu has done a detailed look at the different needs of different regions, and provided a consolidated idea / design for specific tabs, and locations within existing tabs where users will be given the option of adding arbitrary iframes in the backend.

@kamicut , we need to:

 - Do a detailed review of the wireframes above and clarify with @mariam-yu where required to figure out what additional fields we need to add in the backend to allow users to add custom iframes at places, give custom names to tabs, give custom names to certain section headers, etc.
 - Once we have an enumerated list of all the additional fields we need in the admin to define these content blocks that can be defined by site admins, we need to: add the fields to the db and admin and expose them in the API.
 - Once all the custom fields are coming through to the API and frontend, we can work with @imohkay to have everything show up in the correct place and as per the design guidelines.

@kamicut - let's discuss when we next speak.

**Comment by tovari** [comment]
Linking here the ESSN - custom tabs on country pages ticket (https://github.com/IFRCGo/go-frontend/issues/1637) as it is related and can be handled in the same ticket. 
I close that ticket.

**Comment by batpad** [comment]
From going over the wireframes, for regions, these are the additional fields we will need in the Admin backend:

 - `Emergencies` tab: User can add multiple Sections (a section is a Title + Iframe link) that display below the existing GO data under the Emergencies tab. User can use some simple mechanism like assigning an "Order no" to order sections.

 - `Regional Profile` tab: Allow users to add `Additional Links` - Links are Title + Link. User can choose whether links should open in a new tab / window, or open "as an Iframe". @kamicut - let's look at the proposed design for this Additional Links feature and how these proposed "larger" Iframe embeds would work, and check if you see red flags / have an estimation for the work there. Users can additionally add `Sections` for the Regional Profile tab which functions the same as Sections on the Emergencies tab.

 - `Preparedness`: User can add Preparedness Sections on the backend (Title + Iframes) and provide a custom ordering for them.

- `Additional Data` tab: Functions the same as now, user should now be able to give a custom name to the tab to call it something other than `Additional Data`.

The other desired addition is including `Key Documents` in the Regional Profile tab. This would function similar to `Response Documents` on Emergencies. I would recommend doing this as a follow-up after we have all the above working.

@kamicut - will be great to go over these real quick - re-organizing the data into this new tab structure and creating these various customizable embed sections in the admin will be a bit of work, but I think is mostly reasonably straightforward. The thing I am most concerned about is the `Additional Links` function and having these links open up as embedded iframes taking up the entire tab, along-with back button navigation. Let's take a look at the proposed design here and make sure everything seems feasible.

**Comment by batpad** [comment]
@kamicut - for clarity, these are the wireframes: https://invis.io/ZBZB8SF9UCP#/436634656_2-1_GO_Regions_AF - we need to look at the Region pages for Africa and America to get a full picture of the content organization within these various tabs.

**Comment by mariam-yu** [comment]
Posting the link to the updated country pages for Kenya that include the 2 Africa Ops Room addition suggestions: PAC indicators (under Country Profile top left) and WASH capacity (under Preparedness): https://invis.io/XMZE5XXV5HN#/437418443__GO_Country_Op_Kenya

Next steps: double checking the integration with @batpad, creating a common list for all countries for PAC indicators (@mmusori to check during the meeting with regions), and for us to update the Preparedness part to include PER and a better split between preparedness info (capacity, etc.) vs. assessments (PER, OCAC, etc.). Thanks!

**Comment by maisa-yu** [comment]
@batpad   Sharing with you some modules from Regional pages:
- Information Cards 
- Additional Links
- All countries sidebar
We’re working on the complete pages and will share with you next week, so you can focus only on those modules for now.

Here are the links to zeplin screens:
zpl.io/VqpDqvl
zpl.io/bz4K4kM

**Comment by imohkay** [comment]
@maisa-yu  Could you provide the icons for the Information Cards? The ones for National Societies and Country Clusters?
Right now, am unable to download from Zeplin. Thanks!

**Comment by maisa-yu** [comment]
Hi @imohkay the icons are already exportable there, not sure if there's something wrong because they're in a group... it changes depending where you click. 
I'll try exporting again to see if it works.

**Comment by imohkay** [comment]
@maisa-yu I can download the vector file now. Thank you:)

**Comment by batpad** [comment]
The tab re-organization and additional tabs on Region pages is ready to test.

What can be tested on the Region Admin and frontend pages:
 
 - Tabs and organization of sections on the Region page are as per new designs: PENDiNG: List of Countries to be moved to slider.
 - Add a custom name for `Additional Tab Label` in the Region Admin - editing it should change label for Additional Data tab on frontend.
 - Add custom "Emergency Snippets" to show up in the Emergency tab on the Region page
 - Add custom "Profile Snippets" to show up in the Profile tab on the Region page
 - Add custom "Preparedness Snippets" to show up in the Preparedness tab on the Region page (Preparedness tab will not show if there are no Preparedness Snippets).
 - Adds an option to `Show in GO` for Region Links in the Admin. If this option is checked in the Admin, it will display that link within the Regional Profile tab as an iframe.
 - Adds counts for National Societies and Country Clusters inside Region Profile - these values are derived from the database - @jhenshall we may need to discuss whether these will be reliable to use, or if we should hide the cluster count for now.

@karitotp will be great if you can test these. Let me know if anything needs clarification.

@mmusori this is ready for testing on staging - let me know if you are able to have a look and if things make sense. We can likely improve the organization of the forms in the Admin a bit, but hopefully this makes sense for now and that is something we can do down the line.

@mariam-yu ^ this is ready for testing, couple of minor known issues with icons we are fixing, as well as moving the list of countries to the slider.

**Comment by batpad** [comment]
Additionally, we have added the new fields for National Society Indicators and WASH Indicators to the Country admin backend. We have not yet added these to the Country page on the frontend - this is what we will work on next and expect to have it ready by Thursday.

**Comment by batpad** [comment]
@mariam-yu for the WASH capacity indicators and NS Indicators on the country page, will it be on Zeplin soon?

Also, we are still not pushing the Country Profile tab live - the National Society Indicators are currently in that tab. Should we then add these when we go live with the Country Profile / Historic Databank tab? cc @LukeCaley @mmusori on this.

**Comment by karitotp** [comment]
Hey @batpad, I did a test to these functionalities and here my comments:

> - Tabs and organization of sections on the Region page are as per new designs: PENDiNG: List of Countries to be moved to slider.

I found these differences comparing the design in https://projects.invisionapp.com/share/ZBZB8SF9UCP#/screens/436634656_2-1_GO_Regions_AF and the Region page in Staging.

![Selection_010](https://user-images.githubusercontent.com/11504548/100159754-53b31600-2e7c-11eb-869f-57043e1bfe76.png)


> - Add a custom name for Additional Tab Label in the Region Admin - editing it should change label for Additional Data tab on frontend.

This works good, I changed the label name and it reflects on frontend. https://dsgoproxyapp.azurewebsites.net/regions/3#additional-info


> - Add custom "Emergency Snippets" to show up in the Emergency tab on the Region page

Working good: https://dsgoproxyapp.azurewebsites.net/regions/3#operations

![Screenshot from 2020-11-24 17-50-03](https://user-images.githubusercontent.com/11504548/100160397-7f82cb80-2e7d-11eb-9e38-90a6fdac0316.png)


> - Add custom "Profile Snippets" to show up in the Profile tab on the Region page

This works, but I added an image that did not appear in the page: https://dsgoproxyapp.azurewebsites.net/regions/3#regional-profile

This is the warning that show me in admin.
![Screenshot from 2020-11-24 17-57-06](https://user-images.githubusercontent.com/11504548/100160961-8d851c00-2e7e-11eb-87ab-fe4dbb2870f5.png)


> - Add custom "Preparedness Snippets" to show up in the Preparedness tab on the Region page (Preparedness tab will not show if there are no Preparedness Snippets).

Works good :heavy_check_mark: https://dsgoproxyapp.azurewebsites.net/regions/3#preparedness


> - Adds an option to Show in GO for Region Links in the Admin. If this option is checked in the Admin, it will display that link within the Regional Profile tab as an iframe.

I added three links in Admin, and only one was selected.

![Screenshot from 2020-11-24 18-05-18](https://user-images.githubusercontent.com/11504548/100161864-4435cc00-2e80-11eb-94f9-b6fb8394eac7.png)

This works, if I click over the selected one it display the page within the profile page.

![Selection_012](https://user-images.githubusercontent.com/11504548/100163345-7563cb80-2e83-11eb-8d38-40da071fe308.png)

**Comment by batpad** [comment]
@karitotp thank you!!

@imohkay tagging you to rename Operations to Emergencies. I can change the default name of the Additional Data tab.

With regards to uploading images in the editor in the backend, could you provide an exact steps to reproduce? Also, are you able to tell if the same issue occurs across ALL snippet / graphical text editor fields? I imagine that getting image uploading to work in the graphical editor is more work - cc @GregoryHorvath do you know if this is supposed to work / if it maybe better to remove the button for users to upload images into snippets for now? @karitotp if you can give exact steps to reproduce, we can discuss a fix.

Thanks again @karitotp this seems great! As a next step here, could you do some more thorough testing across languages? Entering data in multiple languages in the admin, switching language in the frontend, etc?

@mmusori - from @karitotp's testing, it looks like confirmation that all the major features here seem to work well. We will work on the minor issues identified, but will be great if you all are able to give this a review from your end, and let us know if all this works for users and if there's anything we should discuss.

**Comment by GregoryHorvath** [comment]
@batpad Yeah I've been thinking about removing that separate image upload for Snippets... although I'm not sure if anyone used/uses that actively. I've definitely seen it cause issues "randomly".

**Comment by imohkay** [comment]
@maisa-yu @mariam-yu Hey, regarding the All Countries new UI (with the << All Countries link just above the tabs), is this link placed at the extreme right edge of the browser? Or it sits just above the last tab (Additional Data) on the right?

**Comment by tovari** [comment]
hi @batpad,
looks good!
I'm wondering why we have 2 fields to add the additional tab name:
![image](https://user-images.githubusercontent.com/68283148/100219395-3cc10200-2f16-11eb-8d47-89a5f077dd17.png)

On the Country page admin, I was not able to update the Additional tab name:
![image](https://user-images.githubusercontent.com/68283148/100219688-97f2f480-2f16-11eb-88f4-9e5119176595.png)

![image](https://user-images.githubusercontent.com/68283148/100219761-ad681e80-2f16-11eb-8d22-1d5ae84bc083.png)

https://dsgocdnapi.azureedge.net/en/api/country/174/change/?_changelist_filters=q%3Dturkey
https://dsgoproxyapp.azurewebsites.net/countries/174#additional

Also, the nem figures (WASH, etc) are mandatory now. Is this the intention @mmusori ?

Thanks!

**Comment by mariam-yu** [comment]
@batpad @imohkay Please don't rename the Operations to Emergencies yet. It was not approved. (the wireframes are always a bit ahead with some concepts). As for Additional Information, if you do rename to Additional Data, could you please make it across (including on regions + emergencies).

Lastly, regarding the All Countries, could we have it attached to the browser edge and test it? Would that be an easy thing to change if that comes out strange?

Thanks!

**Comment by imohkay** [comment]
@batpad Please checkout feature/1621-country-sidebar. Could you make the filters for the country sidebar work? Thanks.

**Comment by batpad** [comment]
> lso, the nem figures (WASH, etc) are mandatory now. Is this the intention @mmusori ?

This was a bug, this should now be rectified on staging.

Thanks for the catch about something weird with the Additional Tab Name, possibly when I enabled translations for that field, I will look into it.

**Comment by batpad** [comment]
@mariam-yu @mmusori - the All Countries slider is now on staging to test, eg https://dsgoproxyapp.azurewebsites.net/regions/2#operations

Let us know how this looks!

@karitotp could you also please test the `All Countries` button that appears above the tabs? That it works and that searching within it works.

With this, I think the Region page is complete. I think the only pending thing is figuring out whether we want to hide the Country Cluster info for now while @jhenshall is working on compiling a more complete list.

I have moved the ticket to Ready for QA - will be great to get any feedback / bugs in the next days.

**Comment by batpad** [comment]
> On the Country page admin, I was not able to update the Additional tab name:

@tovari I see the updated tab name here: https://dsgoproxyapp.azurewebsites.net/countries/174#operations - can you confirm that this works or if you are able to reproduce the issue you were having?

For the other issues with the inconsistent label name and the extra field showing up for additional tab for the Region page, thanks for the catches! - Fix is up at https://github.com/IFRCGo/go-api/pull/939 and should be deployed to staging by tomorrow.

**Comment by tovari** [comment]
> > On the Country page admin, I was not able to update the Additional tab name:
> 
> @tovari I see the updated tab name here: https://dsgoproxyapp.azurewebsites.net/countries/174#operations - can you confirm that this works or if you are able to reproduce the issue you were having?
> 
> For the other issues with the inconsistent label name and the extra field showing up for additional tab for the Region page, thanks for the catches! - Fix is up at [IFRCGo/go-api#939](https://github.com/IFRCGo/go-api/pull/939) and should be deployed to staging by tomorrow.

Thanks, @batpad, It works as expected!

**Comment by mariam-yu** [comment]
@batpad and team. First of all great job putting this together. It looks great! Went through it and a few minor comments:

- The sidebar works well. Super excited to see this. Do you guys think it could be possible to have the All Countries button to move along with the sidebar? So basically the sidebar is attached to the browser edge but the button is attached to the sidebar (Zeplin may help). If creates difficulty, we can keep as is, as it already is great!
- On Regional Profile, if we don't have data on cluster or number of NS, can we not show that box or put a dash? 0 is misleading.
- Also on Regional Profile, can we have the header mandatory for the snippets? Looks weird with the line only, or the line goes with the header basically.
- Contacts looks good, just could you check the styling in Zeplin? Not sure if this was one of the modules we had updated in the last round of UIs.
- Additional links: could we have them organized in columns (see Zeplin) and not spelled in caps? Also, when the link is clicked, could it jump to the top of the page?
- The styling of our Back buttons I think are red, but @maisa-yu to confirm.

Thank you!

**Comment by batpad** [comment]
@karitotp could you please test the All Countries button on Region pages, across browsers and maybe some screen-sizes, and provide feedback?

**Comment by batpad** [comment]
> On Regional Profile, if we don't have data on cluster or number of NS, can we not show that box or put a dash? 0 is misleading.

Yes. This is dependent on us having the country cluster data in the database. We will likely need to hide it for this release - cc @imohkay 

> Also on Regional Profile, can we have the header mandatory for the snippets? Looks weird with the line only, or the line goes with the header basically.

I thought the title should be mandatory in the backend, will check, thanks for the catch.

> Contacts looks good, just could you check the styling in Zeplin? Not sure if this was one of the modules we had updated in the last round of UIs.

@imohkay can you check?

> Additional links: could we have them organized in columns (see Zeplin) and not spelled in caps? Also, when the link is clicked, could it jump to the top of the page?

Yea, getting the jump to not be jarring will be interesting, will look at that. For the columns, in Zeplin I believe they had categories that we don't have, but perhaps we can put them in columns anyway - @imohkay ?

> The styling of our Back buttons I think are red, but @maisa-yu to confirm.

@imohkay

**Comment by mariam-yu** [comment]
@batpad sounds good for all. For additional links, we removed the categories, but some column organization will help. Zeplin is updated.

As for mobile, our quick recommendation would be to make it appear on top and expand downwards, but not sure what kind of effort it is. We haven't really tackled mobile; so, not sure how that page will even look.

**Comment by karitotp** [comment]
> @karitotp could you please test the All Countries button on Region pages, across browsers and maybe some screen-sizes, and provide feedback?

It works well to me @batpad, I test it in different screen size and in firefox and chrome browser and I did not find any issues, only I saw one thing:

- If the button is displaying the list of countries and you go to another region page, the button is still active displaying the list of countries. Maybe this list should be inactive per defect if you change to another region?


Other thing (no related to the countries button) but that is not displaying correctly is an icon in the tables:

![Screenshot from 2020-11-27 09-56-22](https://user-images.githubusercontent.com/11504548/100465396-3b6e1180-309d-11eb-8293-ce76f5e78a39.png)

**Comment by batpad** [comment]
@karitotp thank you for the catch around the icons.

These should now be fixed. But it will be great if you can pay some attention to icons throughout the site as we had to change some things with the way we handle icons. Please let us know if you notice anything weird or off with any icons through the site. Thanks!

**Comment by mariam-yu** [comment]
@batpad Not sure of the status of this, but tagging it here anyway, updated designs for NS page with the NS indicators (on country profile): https://zpl.io/a7APyP8 and with Wash stuff on Preparedness: https://zpl.io/b6qkQBR

**Comment by mmusori** [comment]
1.	All countries appearing in the All countries slider (test done for Africa Region)
2.	The search is quick but puts in additional countries not starting with the first two letters but have those letters in the name. (Not a deal breaker).  
3.	Are we missing a Trend Analysis tab or this is now under Operations? Makes sense though. 
4.	Regional Snippets in the backend map onto the additional graphics under Additional information?

5.	I don’t see the functionality for adding “Key Documents.”

**Comment by batpad** [comment]
> The search is quick but puts in additional countries not starting with the first two letters but have those letters in the name. (Not a deal breaker).

Ah, right now the search is setup to search anywhere within the country - this seemed like what we would want, because for eg. if I search for "Congo", I'd probably want it to match Democratic Republic of Congo, but happy to discuss.

> Are we missing a Trend Analysis tab or this is now under Operations? Makes sense though.

@mariam-yu suggested we keep it under Operations for now until we have more things to put into Trend Analysis.

> Regional Snippets in the backend map onto the additional graphics under Additional information?

Yes, the existing snippet behaviour has not been modified in any way. Modifying the existing snippets would involve potentially breaking functionality of all existing snippets. So the existing Region Snippets still map onto Additional Graphics in Additional Information (the tab can now be called something else).

> I don’t see the functionality for adding “Key Documents.”

Adding document uploading was felt out of scope of this phase of additional tabs work, sorry if this wasn't clear. Again, happy to discuss.

**Comment by tovari** [comment]
On the Additional Information tab, there is an Additional Graphics label, which is not editable at the backend. As we can change the tab name, it would make sense to provide an option to update the Additional Graphic text as well. Thanks!

**Comment by mmusori** [comment]
@batpad just a few minor things to look into. 
1.	Regional key figures are in additional tab should they not be in Regional Profile?  (They are coming in correctly) @mariam
2.	Number of NS with ongoing activities is not consistent between the key figure and the drop-down list. The key figure is 14 and the dropdown has over 30 NS. The export has only 9 ongoing projects. but the export has quality issues, the budget column has a mixture of dates and number.  

![image](https://user-images.githubusercontent.com/7780809/100894173-7ff21680-34bc-11eb-98f8-86f8bf392eec.png)

 3.	I believe number of national societies in the region come from GO data base. Thus, when we have the cluster information this will be available too. Current issue is that for Americas/Asia Pacific /MENA/Europe the this figure is mislabeled as National Societies in Africa.
  
![image](https://user-images.githubusercontent.com/7780809/100894213-8a141500-34bc-11eb-9356-0decdfdf566d.png)


4.	Additional links on Africa Regional page not taking you to the content on GitHub. Solutions proposed already. 
5.	Align the backend with the frontend. It should be possible to tell where things end up on the frontend after putting into the backend.

**Comment by batpad** [comment]
Thanks @mmusori !

> Regional key figures are in additional tab should they not be in Regional Profile? (They are coming in correctly) @Mariam

We didn't have a clear space for the existing configurable Key Figures in the Regional Profile - the existing Additional Tab was left as is (other than making the name configurable) - if there's a clear space where the Key Figures should go, happy to speak to @imohkay about making that change. @mariam-yu ?

> Number of NS with ongoing activities is not consistent between the key figure and the drop-down list. The key figure is 14 and the dropdown has over 30 NS. The export has only 9 ongoing projects. but the export has quality issues, the budget column has a mixture of dates and number.

@mmusori are these all issues within the 3w tab? We have not modified anything inside 3w - it would be really great if we could ticket these separately as issues within 3w are definitely not related to the additional tab changes, and ideally would be something tackled separately, I think.

> I believe number of national societies in the region come from GO data base. Thus, when we have the cluster information this will be available too. Current issue is that for Americas/Asia Pacific /MENA/Europe the this figure is mislabeled as National Societies in Africa.

That is correct about cluster information - we will add that box back once we have that information in the daabase. Thank you for the catch about the mislabeling! Shall fix.

> Align the backend with the frontend. It should be possible to tell where things end up on the frontend after putting into the backend.

As discussed, for now we will rename the fields in the backend to give a clearer indication of where the snippets will appear. In the future, we should scope out a restructuring of the Admin form to organize it into clearer sections, etc.

I will fix the mis-labelling of the number of national societies widget, and change the form labels in the Admin. And @mariam-yu - let us know about the existing Key Figures that currently show in Additional Information - there was not a place for it in any of the other tabs, unless we missed something, so we left it in Additional Information - but happy to move it if there's a clear idea where.

Thanks again for the testing and feedback, @mmusori - hopefully all the above sounds good.

**Comment by tovari** [comment]
> On the Additional Information tab, there is an Additional Graphics label, which is not editable at the backend. As we can change the tab name, it would make sense to provide an option to update the Additional Graphic text as well. Thanks!

I suggest to rename the 'Additional Graphics' text to 'Regional content' on the frontend for this release.

**Comment by mmusori** [comment]
Thanks @batpad . 
1. I will wait for the response from @mariam-yu on the Key Regional figures.
2. I will ticket separately the regional 3W issues. 
3. I will test the Regional profile labels and the backend/frontend alignment once it is done.

**Comment by mariam-yu** [comment]
@batpad @mmusori what are these key figures? If it's what, say, Americas has currently in their Additional Info tab, then it's the same as the top numbers (number of NS and clusters per region) in Regional Profile, which is already there. So, I guess that should be removed.

A few minor comments from my side:
- On operations, the Recent Emergencies is breathing down the neck of our trend graphs :P. Could we give it some space? Should be equal across, and yes @mmusori temporarily hosting the trend graphs here, while we figure out how to enrich the Trend tab.
- Regional Profile - the header of the snippet is still missing, shows just a line.
- Thanks for organizing the Additional Links. Looks good. Could we have them all not in caps? We also had some styling for link hovers (across the platform), but not sure if we need to worry about that in this iteration.
- For All countries slider - would it be possible to add some easing in/out and make the transition quicker? May be something like this: https://codepen.io/dannievinther/full/NvZjvz (of course the page shouldn't darken). Also, would it be possible to make the height aligned with the bottom of the browser (?) and, potentially, have the sidebar move with the page (sticky)?. But please let me know if this is all too much work. Thanks!

**Comment by batpad** [comment]
> @batpad @mmusori what are these key figures? If it's what, say, Americas has currently in their Additional Info tab, then it's the same as the top numbers (number of NS and clusters per region) in Regional Profile, which is already there. So, I guess that should be removed.

We still have a section in the Admin allowing users to put in arbitrary Key Figures (figure, deck and source). I'd be a bit worried to remove that entirely in case people are using it.

Am happy to hide it in the frontend for now while we take a decision on whether to remove it entirely from the Admin, or figure out a new location. @mmusori ?

**Comment by batpad** [comment]
> Regional Profile - the header of the snippet is still missing, shows just a line.

@mariam-yu could you screenshot / share link of where you are seeing this? I seem to see the snippet title fine on for eg. http://localhost:3000/regions/0#regional-profile

> Could we have them all not in caps?

Links are displayed as entered in the admin - it can be easily changed to non all caps in the admin.

**Comment by batpad** [comment]
The following frontend fixes are up in a Pull Request and should be on staging soon:

  - Fix region name in label for national societies count
  - Change Additional Graphics to Regional Content in Additional Data tab
  - Hide Key figures
  - Hides the Additional Info tab if it contains no data

@imohkay - can you check the minor styling things @mariam-yu pointed out: I think those are:

 - Spacing between Recent Emergencies and Graphs (see @mariam-yu's comment above)
 - Styling and behaviour of countries slide-in (there are likely limitations to what can be done here, but would be good to see if any of those things are straightforward and maybe outline what's possible and what's not here).

**Comment by batpad** [comment]
@mmusori - have updated the naming for the snippets in the Admin to make it clearer where each snippet goes.

@mariam-yu - @imohkay has addressed some of the styling issues - I think additional issues with the country slider would require some discussion as some of the things are not straightforward. Hopefully is fine as is for this release.

With this, I think we've addressed all the issues identified. Will be great to do a final round of testing and also let us know if anything seems pending before we can release.

**Comment by mmusori** [comment]
@batpad unfortunately the key figures have been hidden in the front end before @mariam-yu could view this. 
in the backend @mariam-yu the input is done here.
![image](https://user-images.githubusercontent.com/7780809/100982681-ac537480-3548-11eb-8a43-04f149ac43c9.png)

**Comment by tovari** [comment]
> > On the Additional Information tab, there is an Additional Graphics label, which is not editable at the backend. As we can change the tab name, it would make sense to provide an option to update the Additional Graphic text as well. Thanks!
> 
> I suggest to rename the 'Additional Graphics' text to 'Regional content' on the frontend for this release.

@batpad, in case of the Country pages, the Additional Graphics text could be removed. I cannot suggest any text that would fit here. 
In case, the string must be the same for Regional and Country pages, it is better to remove it.
Thanks!

**Comment by imohkay** [comment]
@mariam-yu
1. For the country slider, I've reduced the animation speed from 200ms to 100ms. Also, we can't make the height of the dropdown the height of the browser so easily (because we will need to know the scrolled position of the user and also keep it full height as the user scrolls the page. If we had the Countries button always fixed to the page in a corner (like the Codepen demo), it would be a different matter...

**Comment by batpad** [comment]
@mmusori have added back the Key Figures per your comment above.

**Comment by batpad** [comment]
> @batpad, in case of the Country pages, the Additional Graphics text could be removed. I cannot suggest any text that would fit here.

Was the text on Country pages affected? @tovari can you give me an example URL for countries that has snippets? Thanks.

**Comment by mariam-yu** [comment]
@imohkay Thanks, may be we try 150ms? Could we add easing? That would help with the smoothness. Thankssss

**Comment by mariam-yu** [comment]
@batpad Can see the title now, may be was not refreshed. Thank you!!!

**Comment by tovari** [comment]
> > @batpad, in case of the Country pages, the Additional Graphics text could be removed. I cannot suggest any text that would fit here.
> 
> Was the text on Country pages affected? @tovari can you give me an example URL for countries that has snippets? Thanks.

https://dsgoproxyapp.azurewebsites.net/countries/174#additional

**Comment by batpad** [comment]
@tovari is it okay if it says `Country Snippets` or so to remain consistent with region? Removing the title altogether might have other implications to the layout, etc.

**Comment by mmusori** [comment]
@batpad the Number of national societies in the Regional profile display correctly and are now labeled correctly but only in English. The translated pages display 0 national societies.

![image](https://user-images.githubusercontent.com/7780809/101129363-b71e1000-3601-11eb-8d3c-7ffc1f4c97b6.png)

**Comment by batpad** [comment]
> The translated pages display 0 national societies.

Wow, thank you for the catch, @mmusori - that was definitely an interesting bug - to tell whether a Country has a national society, we are checking if the national_society_name field is empty - when switching to other languages, the check would happen on that language specific string :facepalm: - fix for this is up here: https://github.com/IFRCGo/go-api/pull/948 - should be deployed to staging in a bit.

**Comment by tovari** [comment]
> @tovari is it okay if it says `Country Snippets` or so to remain consistent with region? Removing the title altogether might have other implications to the layout, etc.

Hm, this additional tab will be used for publishing various content, so I'm not sure we could have a single string that would fit all. E.g. in case of the ESSN project, Country snippets would not fit. Is it possible to change it to whitespaces? Does it break the layout?

**Comment by batpad** [comment]
@imohkay could we look at what it would take to remove the Additional Graphics title altogether on the Countries page?

**Comment by batpad** [comment]
@tovari have removed the header in the additional tab for countries.

**Comment by batpad** [comment]
> The translated pages display 0 national societies.

This should also be fixed on staging.

**Comment by mmusori** [comment]
The the Key Regional figures are fine in additional information tab for now. I talked to Elly and he only wanted the Key figures from Appeals which is already done automatically.

**Comment by tovari** [comment]
There are a few styling issues on the Arabic page:

- Country slide in button arrow direction should be opposite
- Regional profile - additional links:
  
![image](https://user-images.githubusercontent.com/68283148/101295933-d34bc800-3820-11eb-9365-90f24ddf473b.png)

- Country names on the 3W SAnkey are weird mixture of characters

Additional links on Regional profile don't work for me, and I was not able to load the ifrcgo.org page either. Maybe it has been already shut down?

**Comment by batpad** [comment]
> Country slide in button arrow direction should be opposite

@imohkay would this be possible to fix?

> Regional profile - additional links:

I **think** that would be expected styling for a right-to-left language. @imohkay can you confirm / recommend if there's anything we can do here?

> Country names on the 3W SAnkey are weird mixture of characters

@tovari is this a new issue or does it exist in prod? Will look anyway, but if it's an existing issue, will be great if we can file separately.

> Additional links on Regional profile don't work for me, and I was not able to load the ifrcgo.org page either. Maybe it has been already shut down?

It does seem like the domain is no longer accessible. We will need the new domain setup or the current domain renewed for these to work again.

**Comment by mmusori** [comment]
@batpad The NS figures in the regional profile are now fixed for all regions in the other language's so thanks. The only outstanding issues are:
1. The links to the content on ifrcgo before the ops room went offline it was possible to view the pages in ifrcgo.org but not possible to view on GO. So in the backend if you click on view on GO then you did not get a result. But if you unclick that it was possible to be taken to the website. 
2. The 3W which @batpad  says needs a separate ticket. I will put the issues into a new ticket.

**Comment by batpad** [comment]
> The links to the content on ifrcgo before the ops room went offline it was possible to view the pages in ifrcgo.org but not possible to view on GO. So in the backend if you click on view on GO then you did not get a result. But if you unclick that it was possible to be taken to the website.

Yes, the `Show in GO` check-box signifies whether the link should be embedded as an iframe within the site, or just be treated as a normal external link. When we had it set to the Github Pages URL, the Iframe embed did not work, but you could go directly to the site. So we switched to ifrcgo.org for testing, and now, of course, the domain has gone down.

I believe the domain issues are being resolved. We could also setup the links on the Africa page without the `Show in GO` checkbox, to behave as normal external links while the domain issue gets resolved, and just change it in the backend once it does.

> The 3W which @batpad says needs a separate ticket. I will put the issues into a new ticket.

Would this be a release blocker for this release? I'd be very surprised if it was a new issue introduced by any of the current work.
